packages/x-components/src/wiring/utils/wire-racing-handling.ts

```diff
@@ -134,8 +134,11 @@ function updateRaceSubscription(
  * @param timingOperatorRx - The kind of Rx operator to be checked.
  * @returns If the timing operator provided is a debounce.
  */
-function isDebounceRx(
-  timingOperatorRx: typeof debounceRx | typeof throttleRx
-): timingOperatorRx is typeof debounceRx {
+function isDebounceRx(timingOperatorRx: RXDebounce | RXThrottle): timingOperatorRx is RXDebounce {
   return timingOperatorRx === debounceRx;
 }
+
+/* Type hack to allow TS to infer correct types. If `__kind__` is removed, TS will infer `never` in
+`else` type guard clauses*/
+type RXDebounce = typeof debounceRx & { __kind__?: 'debounce' };
```

Comment:
TS seems to have some serious problems when  using type guards for functions. I don't know how this was working before, I created this playground with the old 3.7.5 ts version and a simple example to illustrate this.

Fortunately, I came up with this hack that seems to do the trick. I don't like it, but It was the only solution that didn't imply changing the code, which was my initial intention as I don't understand why we have to pass the Rx function to use (either `throttle` or `debounce`) to the `handleTimedWire` function, and then this `handleTimedWire` function has two completely different paths for each of them.